t/unit/backends/test_mongodb.py

```diff
@@ -163,6 +164,20 @@ def test_get_connection_no_connection_mongodb_uri(self):
             )
             assert sentinel.connection == connection
 
+    def test_get_connection_no_connection_mongodb_srv_url(self):
+        with patch('pymongo.MongoClient') as mock_Connection:
+            mongodb_uri = self.srv_url
+            self.backend._connection = None
+            self.backend.host = mongodb_uri
+
+            mock_Connection.return_value = sentinel.connection
+
+            connection = self.backend._get_connection()
+            mock_Connection.assert_called_once_with(
+                host=mongodb_uri, **self.backend._prepare_client_options()
```

Comment:
I assumed that `MongoClient` would be initialized with `mongodb://hostname.dom/database`. Isn't that the case?

Reply:
As I said above. MongoClient is passed the full uri unmodified and does all the DNS lookups (SRV, TXT, A).

Reply:
If you look at the test I added to Kombu, you can see what the MongoClient/parse_uri methods do:

https://github.com/celery/kombu/pull/859/files#diff-48b6aab62ddabad370445ef13ab32d41R90

Reply:
It seems like this is still a hangup, so I'll be explicit: MongoClient will fail if you pass it `mongodb://<host>/<db>...` because that hostname may not exist in DNS. One could ALSO add an A record that would get looked up, but MongoClient, when passed `mongodb+srv://<host>/<db>` will look up `SRV _mongodb._tcp.<host>`, and then `TXT <host>` to get additional parameters. The SRV record will point to one or more mongod servers, and will configure the client as `mongodb://<srv1>:<port1>[,<srv2>:<port2>..]/<db>?<TXT options>`, but unless celery does the lookup itself, just stripping off "+srv" before passing it to MongoClient will not work.